runtime/jvmti/jvmtiHook.c

```diff
@@ -2244,18 +2252,20 @@ jvmtiHookClassFileLoadHook(J9HookInterface** hook, UDATA eventNum, void* eventDa
 	J9VMClassLoadHookEvent * data = eventData;
 	J9JVMTIEnv * j9env = userData;
 	jvmtiEventClassFileLoadHook callback = j9env->callbacks.ClassFileLoadHook;
-	J9JavaVM * vm = j9env->vm;
 
 	Trc_JVMTI_jvmtiHookClassFileLoadHook_Entry();
-
+#if JAVA_SPEC_VERSION >= 9
+	J9JavaVM * vm = j9env->vm;
```

Comment:
This won't compile: the declaration of `vm` cannot occur after the call to `Trc_JVMTI_jvmtiHookClassFileLoadHook_Entry`.